x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/AbstractEnrichProcessor.java

```diff
@@ -188,8 +191,9 @@ int getMaxMatches() {
     }
 
     private static BiConsumer<SearchRequest, BiConsumer<SearchResponse, Exception>> createSearchRunner(Client client) {
+        Client originClient = new OriginSettingClient(client, ENRICH_ORIGIN);
         return (req, handler) -> {
-            client.execute(
+            originClient.execute(
```

Comment:
I think this one is right since this is used to access the enrich system indices.

Reply:
++, this one I'm pretty confident on being the right thing.